test/unit/model/DeliveryExecutionFinderServiceTest.php

```diff
@@ -0,0 +1,121 @@
+            ServiceProxy::SERVICE_ID => $this->executionServiceProxy
+        ]));
+    }
+
+    public function testFindDeliveryExecutionByExecutionId()
+    {
+        $sid = 'v5ba19e6ltos1lmljfv8fgnb07:::S3294476:::29123:::dyJ86SiwwA9';
+        $eid = 'http://selor.docker/tao.rdf#i1562270728176451';
+
+        $launchData = new LtiLaunchData([DeliveryExecutionFinderService::LTI_SOURCE_ID => $sid], ['execution' => $eid]);
+
+        /** @var DeliveryExecutionInterface|PHPUnit_Framework_MockObject_MockObject $implementation */
+        $implementation = $this->getMock(DeliveryExecutionInterface::class);
+        $implementation->method('getIdentifier')->willReturn($eid);
+
+        $this->ltiResultAliasStorage->method('getDeliveryExecutionId')->willReturn($eid);
+        $this->ltiLaunchDataService->method('findDeliveryExecutionFromLaunchData')->willReturn(new core_kernel_classes_Resource($eid));
+        $this->executionServiceProxy->method('getDeliveryExecution')->willReturn(new DeliveryExecution($implementation));
+
+        /** @var DeliveryExecution $deliveryExecution */
+        $deliveryExecution = $this->subject->findDeliveryExecution($launchData);
+
+        $this->assertInstanceOf(DeliveryExecution::class, $deliveryExecution);
+        $this->assertEquals($eid, $deliveryExecution->getIdentifier());
+    }
+
+    public function testFindDeliveryExecutionByLisResultSourceId()
+    {
+        $sid = 'v5ba19e6ltos1lmljfv8fgnb07:::S3294476:::29123:::dyJ86SiwwA9';
+        $eid = 'http://selor.docker/tao.rdf#i1562270728176451';
+
+        $launchData = new LtiLaunchData([DeliveryExecutionFinderService::LTI_SOURCE_ID => $sid], []);
+
+        /** @var DeliveryExecutionInterface|PHPUnit_Framework_MockObject_MockObject $implementation */
+        $implementation = $this->getMock(DeliveryExecutionInterface::class);
+        $implementation->method('getIdentifier')->willReturn($eid);
+
+        $this->ltiResultAliasStorage->method('getDeliveryExecutionId')->willReturn($eid);
+        $this->ltiLaunchDataService->method('findDeliveryExecutionFromLaunchData')->willReturn(null);
+        $this->executionServiceProxy->method('getDeliveryExecution')->willReturn(new DeliveryExecution($implementation));
+
+        /** @var DeliveryExecution $deliveryExecution */
+        $deliveryExecution = $this->subject->findDeliveryExecution($launchData);
+
+        $this->assertInstanceOf(DeliveryExecution::class, $deliveryExecution);
+        $this->assertEquals($eid, $deliveryExecution->getIdentifier());
+    }
+
+    public function testNotFoundDeliveryExecution()
+    {
+        $sid = 'v5ba19e6ltos1lmljfv8fgnb07:::S3294476:::29123:::dyJ86SiwwA9';
+        $eid = 'http://selor.docker/tao.rdf#i1562270728176451';
+
+        $launchData = new LtiLaunchData([DeliveryExecutionFinderService::LTI_SOURCE_ID => $sid], ['execution' => $eid]);
+
+        $this->ltiResultAliasStorage->method('getDeliveryExecutionId')->willReturn(null);
```

Comment:
You don't have to specify null return values, it is the default behaviour. You can simple remove this line